src/sage/structure/coerce_actions.pyx

```diff
@@ -224,15 +241,15 @@ def detect_element_action(Parent X, Y, bint X_on_left, X_el=None, Y_el=None):
         except CoercionException as msg:
             _record_exception()
 
-    # element x defining _act_on_
-    try:
-        if x._act_on_(y, X_on_left) is not None:
-            return ActOnAction(X, Y, X_on_left, False)
```

Comment:
I 'traced' (using `print`)
```
sage: cm = get_coercion_model()
sage: c = cm.get_action(ZZ, list, operator.mul)
```
and line 230 yields
```
Left action by Integer Ring on <class 'list'>
```

Reply:
Yea, I think we are just being too strict with making sure `Y` is a parent. We need to allow that to not be a `Parent`, and maybe need to catch an error (or fix some other code). Not 100% sure at this point.

Reply:
I think, to be of any help I would need to know precisely what `detect_element_action` is supposed to be doing.  And, I guess, that's part of the problem, right?

Reply:
Indeed, we are kind of guessing at its purpose based on its code, its doc, and how it is used. Unfortunately there doesn’t seem to be any consistency with these three. @nbruin Do you have any recollections, thoughts, or opinions on this?

Reply:
I'm not sure which three routines are in play here. I don't think we can have an action of integers on a list, if that's what you're aiming for. Lists are way to coarse a "parent" to determine actions for. There may also be the problem that non-parents probably fail to have the right slots/attributes to properly participate in the coercion system. And there's the conflict  that `3*[1,2,3]` and `[1,2,3]*3` already means "repeated concatenation" in python.

There was recently a discussion on action discovery concerning `int` vs. `Integer` where `Parent` participating in actions also came up: https://groups.google.com/g/sage-devel/c/RvmljIs7wBI/
and in a different form in this relatively recent PR: https://github.com/sagemath/sage/pull/37369
(which  in the end did end up with a patch so that action discovery for int on ZZ-modules should also work!)